test/jasmine/tests/axes_test.js

```diff
@@ -5196,6 +5196,243 @@ describe('Test axes', function() {
             });
         });
     });
+
+    describe('label positioning using *ticklabelmode*: "period"', function() {
+        var gd;
+
+        beforeEach(function() {
+            gd = createGraphDiv();
+        });
+
+        afterEach(destroyGraphDiv);
+
+        function _assert(msg, exp) {
+            var labelPositions = gd._fullLayout.xaxis._vals.map(function(d) { return d.periodX; });
```

Comment:
Very nice tests! But can we use `_fullLayout.xaxis.c2d(d.periodX)` to get these back into date string form - eg `1562079600000` would give `'2019-07-02 15:00'`, and it's not clear to me how we got that precise value, but at least it has a recognizable meaning.

Reply:
Also perhaps we can test the text that's stored in `xaxis._vals[].text` so it's obvious what labels we're showing at what datetime values?

Reply:
> Very nice tests! But can we use `_fullLayout.xaxis.c2d(d.periodX)` to get these back into date string form - eg `1562079600000` would give `'2019-07-02 15:00'`, and it's not clear to me how we got that precise value, but at least it has a recognizable meaning.

Good call. Done in https://github.com/plotly/plotly.js/pull/5065/commits/f0f078a17532748e7164e6930bd5f1c9eb2b49ee.

Reply:
> Also perhaps we can test the text that's stored in `xaxis._vals[].text` so it's obvious what labels we're showing at what datetime values?

Done in https://github.com/plotly/plotly.js/pull/5065/commits/90ee834ebb929d39ab4e893edf30c32f8f64eae3.